processor/processor.go

```diff
@@ -2566,10 +2566,13 @@ func (proc *Handle) transformSrcDest(
 	}
 }
 
-func (proc *Handle) saveDroppedJobs(failedJobs []*jobsdb.JobT, tx *jobsdb.Tx) error {
-	if len(failedJobs) > 0 {
+func (proc *Handle) saveDroppedJobs(droppedJobs []*jobsdb.JobT, tx *jobsdb.Tx) error {
+	if len(droppedJobs) > 0 {
+		for i := range droppedJobs { // each dropped job should have a unique jobID in the scope of the batch
+			droppedJobs[i].JobID = int64(i)
```

Comment:
should we make a copy of the `jobsdb.JobT` before setting the new `i` ?

Not sure how those `droppedJobs` will be used outside of this method

Reply:
Looking further at the code I understand that those jobs have not been inserted to jobsdb and thus don't have a jobID, is this the case ?

Reply:
No, the only purpose of these dropped jobs is to help in capturing these stats. These are not actual jobsdb jobs, they get created [here](https://github.com/rudderlabs/rudder-server/blob/master/processor/events_response.go#L55) and never end up in any jobsdb